src/disp/syscall.c

```diff
@@ -1270,6 +1270,67 @@ static MVMDispSysCall async_unix_listen = {
     .expected_concrete = { 1, 1, 1, 1, 0 },
 };
 
+/* handle-mode-write-only */
+static void handle_mode_write_only_impl(MVMThreadContext *tc, MVMArgs arg_info) {
+    MVMOSHandle *oshandle  = (MVMOSHandle *)get_obj_arg(arg_info, 0);
+
+    if (! sizeof(oshandle->body.data) == sizeof(MVMIOFileData))
```

Comment:
That's not a good way for checking the handle type. In fact I'm quite surprised that it works at all. You are after all comparing the size of a `void *` with that of the MVMIOFileData struct.

I think that struct's definition should stay in syncfile.c and all access to it should go through proper accessor functions. Then the check here can simply be whether such a function exists or not.

Reply:
Strange, I must not have pushed this fix, I know I dropped this check locally.